### ISSUE TITLE
Reorganize code

### ISSUE THREAD

**Issue by tungs** [tungs/timesnap#9]
This issue is to discuss reorganizing and reformatting code. If you think there should be changes in code organization/eslint rules/style/markdown/etc. make a case for them here. I'll try to minimize the number of reorganizations, so accepted suggestions may take some time to be integrated.

**Comment by tungs** [comment]
* Split code into separate files, particularly `overwriteTime`, and `overwriteRandom`
* Update ESLint and its pertinent rules/formatting

**Comment by yarres** [comment]
A few elements:
* use esm to allow use of import, export, etc.
* move to async / await could be interesting

**Comment by Asheboy** [comment]
@tungs I have started doing some upgrades:

- async/await
- arrow functions
- consts & lets

https://github.com/Asheboy/timesnap/tree/feature/code-base-upgrade

**Comment by dcworldwide** [comment]
```
async/await
arrow functions
consts & lets
```

That's great... will make the code much easier to read. Promises are a bit messy

**Comment by tungs** [comment]
Agreed that `async`/`await` results in cleaner code, especially for loops. I was holding off because I wanted to keep maximum compatibility with puppeteer (which supported down to version 6.4.0 of node, which doesn't support `async`/`await`), but now more recent releases of puppeteer require node 8.9.0 or higher. I'll take a look into it for the next significant release.

I view `arrow functions`, `const`s, and `let`s as less important, but I'll probably be migrating over to them at some point for node code. There's also destructuring that would be cleaner as well.

**Comment by tungs** [comment]
As an update, [version 0.3.0](https://github.com/tungs/timesnap/releases/tag/v0.3.0) migrated most of the code from Promise syntax to `async` / `await`.
